src/unittests/f32_interval.spec.ts

```diff
@@ -0,0 +1,439 @@
+    { bounds: [-5, 10], value: -10, expected: false },
+
+    // Point
+    { bounds: [0, 0], value: 0, expected: true },
+    { bounds: [0, 0], value: 10, expected: false },
+    { bounds: [0, 0], value: -1000, expected: false },
+    { bounds: [10, 10], value: 10, expected: true },
+    { bounds: [10, 10], value: 0, expected: false },
+    { bounds: [10, 10], value: -10, expected: false },
+    { bounds: [10, 10], value: 11, expected: false },
+
+    // Upper infinity
+    { bounds: [0, kValue.f32.infinity.positive], value: kValue.f32.positive.min, expected: true },
+    { bounds: [0, kValue.f32.infinity.positive], value: kValue.f32.positive.max, expected: true },
+    { bounds: [0, kValue.f32.infinity.positive], value: kValue.f32.infinity.positive, expected: true },
+    { bounds: [0, kValue.f32.infinity.positive], value: kValue.f32.negative.min, expected: false },
+    { bounds: [0, kValue.f32.infinity.positive], value: kValue.f32.negative.max, expected: false },
+    { bounds: [0, kValue.f32.infinity.positive], value: kValue.f32.infinity.negative, expected: false },
+
+    // Lower infinity
+    { bounds: [kValue.f32.infinity.negative, 0], value: kValue.f32.positive.min, expected: false },
+    { bounds: [kValue.f32.infinity.negative, 0], value: kValue.f32.positive.max, expected: false },
+    { bounds: [kValue.f32.infinity.negative, 0], value: kValue.f32.infinity.positive, expected: false },
+    { bounds: [kValue.f32.infinity.negative, 0], value: kValue.f32.negative.min, expected: true },
+    { bounds: [kValue.f32.infinity.negative, 0], value: kValue.f32.negative.max, expected: true },
+    { bounds: [kValue.f32.infinity.negative, 0], value: kValue.f32.infinity.negative, expected: true },
+
+    // Full infinity
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: kValue.f32.positive.min, expected: true },
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: kValue.f32.positive.max, expected: true },
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: kValue.f32.infinity.positive, expected: true },
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: kValue.f32.negative.min, expected: true },
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: kValue.f32.negative.max, expected: true },
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: kValue.f32.infinity.negative, expected: true },
+    { bounds: [kValue.f32.infinity.negative, kValue.f32.infinity.positive], value: Number.NaN, expected: true },
+
+    // Upper edge
+    { bounds: [0, kValue.f32.positive.max], value: kValue.f32.positive.min, expected: true },
+    { bounds: [0, kValue.f32.positive.max], value: kValue.f32.positive.max, expected: true },
+    { bounds: [0, kValue.f32.positive.max], value: kValue.f32.infinity.positive, expected: true },
```

Comment:
Somewhat surprising.
This is because infinity can be treated as the highest finite f32 value?
A comment would be good.

Reply:
Perfect, thanks!